tests/chainer_tests/test_variable.py

```diff
@@ -285,6 +286,35 @@ def test_uninitialized(self):
         assert a.xp is np
         assert a._has_chainerx_array is False
 
+    def check_array(self, xp, x, y, requires_grad=True):
```

Comment:
Some suggestions for the test:
* We should test for every possible pairs of devices. How about making a separate `TestCase` and parameterizing using two `inject_backend_config`s?
* The code inside `pyteset.raises` should be minimal. Enclosing the entire call of `check_array` could make the test pass unexpectedly by an assertion failure from some unexpected place.